app/src/main/java/com/example/drawingApp/ColorPickerUtility.kt

```diff
@@ -0,0 +1,222 @@
+package com.example.drawingApp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.widget.TextView
+import androidx.core.graphics.toColor
+import androidx.core.widget.addTextChangedListener
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import kotlin.math.max
+import kotlin.math.min
+
+object ColorPickerUtility {
+
+    @SuppressLint("ClickableViewAccessibility")
+    fun colorPickerSheet(
+        context: Context,
+        /**
+         * Called whenever the hue is updated via text or slider.
+         * Use it to set wherever you are storing your hue
+         * and then update your color
+         */
+        onHueUpdate: (Float) -> Unit,
+        /**
+         * Called whenever the Saturation or Value is updated via the gradient square
+         * or via text. use it to set wherever you are storing your saturation AND value
+         * update your color afterward
+         */
+        onSatOrValUpdate: (Float, Float) -> Unit,
+        /**
+         * Called whenever text needs to be updated, meaning on any change, hsv, rgb, or text.
+         * Used to give the information of hsv and previous color.
+         * if there was no previous color pass it black. This is used to help
+         * populate the sheets text views and update the slider and gradient indicators as well as set the previous color box.
+         */
+        onColorTextForceUpdate: () -> DrawingViewModel.ColorPack,
+        /**
+         * Called whenever one of the RGB texts is changed by focus by a user.
+         * it passes a ColorInt as the first parameter, used to decide what value to set.
+         * the second int is the actual value you will set your R,G,or B value to.
+         * Color.RED corresponds to red,Color.GREEN to green, BLUE to blue
+         */
+        onRGBTextChange: (Int, Int) -> Unit,
```

Comment:
I was able to handle this locally in the view and didn't use a lambda for it. I would consider handling it that way and only using lambdas to provide information to the viewModel to hold onto.

For example, I only currently have one Lambda in the dialog, and it is to provide the hex string to the viewModel, and on instantiation in the onUpdate, I give it the current hex string in the viewModel, and default that string to #ff0000